### ISSUE TITLE
[Enhancement] A collection of pre-included plugins

### ISSUE THREAD

**Issue by conorvenus** [ciscoheat/sveltekit-rate-limiter#12]
What do you think about the potential of having some premade plugins included in the library? For example, if people are serving their website via CF or another reverse proxy, the clients' IP addresses are obscured and so you have to use the `CF-Connecting-IP` header, otherwise then you just end up ratelimiting CF's IP (e.g., every single user who wants to access that endpoint is now ratelimited).

```ts
export class CloudflareRateLimiter implements RateLimiterPlugin {
    readonly rate: Rate;

    constructor(rate: { limit: number, duration: RateUnit }) {
        this.rate = [rate.limit, rate.duration];
    }

    async hash(event: RequestEvent) {
        return event.request.headers.get('cf-connecting-ip') || event.getClientAddress();
    }
}```

**Comment by ciscoheat** [comment]
Good idea, I'll add this one to a limiters directory in the next release, then you can make a PR there if you want to add more.

**Comment by thethomaswagner** [comment]
There is no need for this: https://svelte.dev/docs/kit/adapter-node#Environment-variables-ADDRESS_HEADER-and-XFF_DEPTH
This should be deprecated.

> The [RequestEvent](https://svelte.dev/docs/kit/@sveltejs-kit#RequestEvent) object passed to hooks and endpoints includes an event.getClientAddress() function that returns the client's IP address. By default this is the connecting remoteAddress. If your server is behind one or more proxies (such as a load balancer), this value will contain the innermost proxy's IP address rather than the client's, so we need to specify an ADDRESS_HEADER to read the address from:

```
ADDRESS_HEADER=cf-connecting-ip node build
```
